src/GraphQL/ResolveFieldContext/ResolveFieldContextExtensions.cs

```diff
@@ -105,12 +105,12 @@ public static IResolveEventStreamContext<T> As<T>(this IResolveEventStreamContex
             if (context == null)
                 throw new ArgumentNullException(nameof(context));
 
-            if (context.Extensions == null || context.Extensions.Count == 0)
+            if (context.OutputExtensions == null || context.OutputExtensions.Count == 0)
```

Comment:
`SetExtension` is only for output extensions, OK. But `GetExtension` may be used for both input and output. I'm not sure about how it may look for public API. Also you may want to change xml comments to reflect changes.

Reply:
How about if we just rename the methods to GetOutputExtension and SetOutputExtension ?

Reply:
`GetInputExtension` should be too.

Reply:
There is no need for a thread-safe `GetInputExtension` method.

Reply:
I'm talking not exactly about **thread-safe** method. I'm talking about such method as is. No one wants to bother with all that stuff with path splitting.

Reply:
Good idea for a subsequent PR.

Reply:
#3615